[Clang][Driver] Fix `--save-temps` for OpenCL AoT compilation

shiltian opened the pull request:
We can directly call `clang -c -x cl -target amdgcn -mcpu=gfx90a test.cl -o test.o`
to compile an OpenCL kernel file. However, when `--save-temps` is enabled, it doesn't
work because the preprocessed file (`.i` file)  is taken as C source file when it
is fed to the front end, thus causing compilation error because those OpenCL keywords
can't be recognized. This patch fixes the issue.

shiltian commented:
I'm working on a test case.

jhuber6 reviewed (commented):
You should add a test that checks the output of `-ccc-print-phases` and `-ccc-print-bindings`. There's some similar cases in `openmp-offload-gpu.c` tests and similar. Also use `CHECK-NEXT` around the following lines.

shiltian committed:
[Clang][Driver] Fix `--save-temps` for OpenCL AoT compilation

We can directly call `clang -c -x cl -target amdgcn -mcpu=gfx90a test.cl -o test.o`
to compile an OpenCL kernel file. However, when `--save-temps` is enabled, it doesn't
work because the preprocessed file (`.i` file)  is taken as C source file when it
is fed to the front end, thus causing compilation error because those OpenCL keywords
can't be recognized. This patch fixes the issue.
